crypto/keyring/keyring.go

```diff
@@ -392,33 +378,41 @@ func (ks keystore) SaveLedgerKey(uid string, algo SignatureAlgo, hrp string, coi
 		return nil, err
 	}
 
-	return ks.writeLedgerKey(uid, priv.PubKey(), *hdPath, algo.Name())
+	apk, err := codectypes.NewAnyWithValue(priv.PubKey())
+	if err != nil {
+		return nil, err
+	}
+
+	return ks.writeLedgerKey(uid, apk, hdPath)
```

Comment:
why we require algo argument in `writeLedgerKey` ?

Reply:
check other comments. We need it for the exported key header. But we don't need that parameter in `writeLedgerKey` because we have `Type()` method both in private key and public key.